### PR TITLE
Making sure we are not loosing focus in service form network tab

### PR DESCRIPTION
**Pull request by mlunoe**
~~Related PR: https://github.com/dcos/dcos-ui/pull/561~~

~~This PR doesn't solve the whole problem, but makes it a little better, i.e. only happen for one field~~

This PR fixes a small issue, user would loose focus in the create service form when entering new ports.

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @jfurrow

File: src/js/components/ServiceForm.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ const DUPLICABLE_FIELDS_TO_WATCH = {
   },
   ports: {
     fields: ['discovery', 'protocol'],
-    blurOnly: ['name', 'lbPort'],
+    blurOnly: ['name'],
```
Comment (jfurrow) [inline src/js/components/ServiceForm.js]: Could you add a short explanation as to why we don't need `lbPort` in this array? To be honest I don't really know what it's doing anyway...

**Comment by mlunoe** [comment]
As it turns out, we need to watch lbPort after all

**Comment by mlunoe** [comment]
ping @daiyiwhale!

**Comment by mlunoe** [comment]
Removing Review label after changes

**Comment by daiyiwhale** [comment]
Ahh you did it! amazing

**Comment by daiyiwhale** [comment]
I am receiving this warning: `The specified value "Not Enabled" is not a valid number. The value must match to the following regular expression: -?(\d+|\d+\.\d+|\.\d+)([eE][-+]?\d+)?` when `Discovery` is not checked.

**Comment by jfurrow** [comment]
I'm receiving the same warning:

**Comment by orlandohohmeier** [comment]
I'm getting the same warning. It seems like we're not toggling the input type properly. Aside from that it works great - awesome job!

**Comment by mlunoe** [comment]
Yes, I am looking into that. Also, there is some mysterious going on with the removal of 😞
Removing the first row, will remove all.

File: src/js/schemas/service-schema/Networking.js [hunk 2]
```diff
@@ -83,11 +83,9 @@ const Networking = {
         });
       },
       filterProperties(service = {}, instanceDefinition, model) {
-        let properties = Networking
-          .properties
-          .ports
-          .itemShape
-          .properties;
+        let {properties} = Networking.properties.ports.itemShape;
```
Comment (mlunoe) [inline src/js/schemas/service-schema/Networking.js]: All in this file is basically just making it easier to understand what is going on

**Comment by mlunoe** [comment]
The warning problem is now fixed, ref: https://github.com/dcos/dcos-ui/pull/916#issuecomment-239280350

**Comment by mlunoe** [comment]
Tests fixed

**Comment by mlunoe** [comment]
ping @jfurrow

File: src/js/components/SchemaForm.js [hunk 3]
```diff
@@ -99,12 +99,14 @@ class SchemaForm extends mixin(StoreMixin, InternalStorageMixin) {
...
-  handleAddRow(prop, definition, newDefinition) {
+  handleAddRow(prop, definition, newDefinition, index) {
     let propID = Util.uniqueID(prop);
     newDefinition = FormUtil.getMultipleFieldDefinition(
       prop,
       propID,
-      newDefinition
+      newDefinition,
+      undefined,
+      index
```
Comment (jfurrow) [inline src/js/components/SchemaForm.js]: In a separate PR, I would love for us to pass an object with these properties, rather than carefully construct this list of arguments each time we call the `#getMultipleFieldDefinition` method. But again, that's not in scope of this PR!

File: src/js/utils/__tests__/FormUtil-test.js [hunk 4]
```diff
@@ -21,21 +21,25 @@ describe('FormUtil', function () {
...
+      expect(result[1].name).toEqual('variable[2].value');
     });
 
     it('does not modify the definition', function () {
       let result = FormUtil.getMultipleFieldDefinition(
         'variable',
         1,
-        this.definition
+        this.definition,
+        null,
```
Comment (jfurrow) [inline src/js/utils/__tests__/FormUtil-test.js]: In `SchemaForm` we pass `undefined` to this method, whereas here we pass `null`. It looks like they both produce the same result, so I would prefer to be consistent (and `null` makes more sense than `undefined` to me)

**Comment by jfurrow** [comment]
@mlunoe This seems to work. I'm struggling a bit to understand the logic but the confusion wasn't introduced in this PR. I left a minor comment, but otherwise it's good to go!

File: src/js/utils/__tests__/FormUtil-test.js (hunk 4 above)
Comment (mlunoe) [inline src/js/utils/__tests__/FormUtil-test.js]: Done!

File: src/js/components/SchemaForm.js (hunk 3 above)
Comment (mlunoe) [inline src/js/components/SchemaForm.js]: Yeah, let's wait with that to the refactor. I don't want to mess around more with this than I actually have to

**Comment by mlunoe** [comment]
@jfurrow Both @orlandohohmeier and I have tested this fairly throughly. The logic _is_ really hard to follow, and it took me a while to get there. I have tried to add descriptions where I felt they were needed.

**Comment by jfurrow** [comment]
Thanks @mlunoe! Looks good.
